src/Components/SignatureModal/SignatureModal.tsx

```diff
@@ -49,25 +49,42 @@ class SignatureModal extends React.PureComponent<Props> {
   }
 
   private renderButtons = () => {
-    const theme = this.state.colors[this.props.theme];
-    const isDisabled = this.state.isSignSubmitted || !this.state.signatureData;
-    const submitButtonTextColor = isDisabled ? theme.palette.lightGray : theme.themeColor;
     return (
       <View style={styles.buttonsContainer}>
-        <TransparentButtonWithChildren onPress={this.resetWebView} width={60}>
-          <Text style={[styles.buttonText, { color: theme.defaultText, width: 60 }]}>
-            {this.props.cancelButtonLabel.toUpperCase()}
-          </Text>
-        </TransparentButtonWithChildren>
-        <TransparentButtonWithChildren disabled={isDisabled} onPress={this.sendSignature} width={40}>
-          <Text style={[styles.buttonText, { color: submitButtonTextColor, width: 40 }]}>
-            {this.props.submitButtonLabel.toUpperCase()}
-          </Text>
-        </TransparentButtonWithChildren>
+        {this.state.isSignSubmitted ? this.renderActivityIndicator() : this.renderCancelButton()}
+        {this.renderSubmitButton()}
       </View>
     );
   };
+  private renderActivityIndicator = () => (
+    <View style={styles.activityIndicatorContainer}>
+      <ActivityIndicator />
+    </View>
+  );
+
+  private renderCancelButton = () => {
+    const theme = this.state.colors[this.props.theme];
+    return (
+      <TransparentButtonWithChildren onPress={this.resetWebView} width={60}>
+        <Text style={[styles.buttonText, { color: theme.defaultText, width: 60 }]}>
+          {this.props.cancelButtonLabel.toUpperCase()}
```

Comment:
Need to make `cancelButtonLabel` prop not a `string`, but `React.ReactNode`. Because:

- most of i18n react libs does not allow you to pass a string easily.
- user can pass text with formatting here.

But you can not use `toUpperCase` then, cause this method exists on string only.

So we need to use `textTransform` in styles instead of `toUpperCase`